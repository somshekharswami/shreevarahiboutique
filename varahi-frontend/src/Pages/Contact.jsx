import React, { useState } from "react";
import { useForm } from "react-hook-form";
import api from "../api/api";

const ContactPage = () => {
  const [buttonDisabled, setButtonDisabled] = useState(false);
  const {
    register,
    handleSubmit,
    reset,
    formState: { errors, isSubmitSuccessful },
  } = useForm();

  const onSubmit = async (data) => {
    setButtonDisabled(true);

    try {
      const response = await api.post("/api/contact", data);
      if (response.data.success) {
        reset();
      } else {
        console.error("Email failed to send:", response.data.message);
      }
    } catch (error) {
      console.error(
        "Error sending email:",
        error.response?.data || error.message
      );
    } finally {
      setTimeout(() => setButtonDisabled(false), 3000);
    }
  };

  return (
    <section className="min-h-screen bg-gradient-to-b from-pink-50 to-purple-50 py-12 px-6 md:px-20">
      <div className="max-w-3xl mx-auto bg-white p-8 rounded-xl shadow-lg">
        {isSubmitSuccessful && (
          <div className="bg-green-50 border border-green-200 text-green-700 px-6 py-4 rounded-md text-center mb-6 shadow-sm animate-fade-in">
            <p className="font-semibold text-lg mb-1">
              Thank you for reaching out ðŸ’Œ
            </p>
            <p className="text-sm">
              Your message has been wrapped in silk and sent to our team. We'll
              get back to you within 2â€“3 business days.
            </p>
            <p className="text-xs mt-2 italic text-gray-500">
              â€“ With love, Varahi Boutique
            </p>
          </div>
        )}

        <form onSubmit={handleSubmit(onSubmit)} className="space-y-6">
          {/* Name */}
          <input
            type="text"
            {...register("name", { required: "Name is required" })}
            placeholder="Enter your full name"
            aria-label="name"
            className="w-full border border-gray-300 px-4 py-2 rounded-md focus:ring-2 focus:ring-[#B19A99] outline-none font-mono"
          />
          {errors.name && (
            <p className="text-red-500 text-sm mt-1">{errors.name.message}</p>
          )}

          {/* Email */}
          <input
            type="email"
            {...register("email", {
              required: "Email is required",
              pattern: {
                value: /^\S+@\S+$/i,
                message: "Invalid email format",
              },
            })}
            placeholder="abc@gmail.com"
            aria-label="email"
            className="w-full border border-gray-300 px-4 py-2 rounded-md focus:ring-2 focus:ring-[#B19A99] outline-none font-mono"
          />
          {errors.email && (
            <p className="text-red-500 text-sm mt-1">{errors.email.message}</p>
          )}

          {/* Subject */}
          <input
            type="text"
            {...register("subject")}
            aria-label="subject"
            placeholder="Let us know why you're reaching out"
            className="w-full border border-gray-300 px-4 py-2 rounded-md focus:ring-2 focus:ring-[#B19A99] outline-none font-mono"
          />

          {/* Message */}
          <textarea
            rows="5"
            {...register("message", { required: "Message is required" })}
            placeholder="Briefly describe.."
            aria-label="message"
            className="w-full border border-gray-300 px-4 py-2 rounded-md resize-none focus:ring-2 focus:ring-[#B19A99] outline-none font-mono"
          />
          {errors.message && (
            <p className="text-red-500 text-sm mt-1">
              {errors.message.message}
            </p>
          )}

          {/* Submit Button */}
          <div className="text-center">
            <button
              type="submit"
              aria-label="submit"
              disabled={buttonDisabled}
              className="bg-[#B19A99] text-white px-6 py-2 rounded-md hover:bg-[#a88a89] transition duration-300"
            >
              {buttonDisabled ? "Sending..." : "Send Message"}
            </button>
          </div>
        </form>
      </div>
    </section>
  );
};

export default ContactPage;
