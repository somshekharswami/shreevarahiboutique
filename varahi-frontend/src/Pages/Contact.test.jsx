import React from "react";
import { render, screen, waitFor } from "@testing-library/react";
import Contact from "./Contact";
import userEvent from "@testing-library/user-event";
import axios from "axios";

// mock axios
jest.mock("axios");

describe("Contact Page Form", () => {
  test("renders all fields and submits successfully", async () => {
    render(<Contact />);

    // Use placeholder text to get form fields
    const nameInput = screen.getByPlaceholderText(/enter your full name/i);
    const emailInput = screen.getByPlaceholderText(/abc@gmail.com/i);
    const subjectInput = screen.getByPlaceholderText(
      /let us know why you're reaching out/i
    );
    const messageInput = screen.getByPlaceholderText(/briefly describe../i);
    const submitButton = screen.getByRole("button", { name: /submit/i });

    // Simulate user typing
    await userEvent.type(nameInput, "Soham");
    await userEvent.type(emailInput, "test@abcgmail.com");
    await userEvent.type(subjectInput, "Subject of the Problem");
    await userEvent.type(messageInput, "this is the description of problem");

    //  Assert inputs received values before submission
    expect(nameInput).toHaveValue("Soham");
    expect(emailInput).toHaveValue("test@abcgmail.com");
    expect(subjectInput).toHaveValue("Subject of the Problem");
    expect(messageInput).toHaveValue("this is the description of problem");

    // Mock successful axios response
    axios.post.mockResolvedValue({
      data: {
        success: true,
        message: "thankyou for reaching out",
      },
    });

    // Submit form
    await userEvent.click(submitButton);

    // Expect success message to appear-//WaitFor :Waits for an async update (like after an API call)
    await waitFor(() => {
      expect(
        screen.getByText((content) =>
          content.includes("Thank you for reaching out")
        )
      ).toBeInTheDocument();
    });
    // âœ… Optional: Assert correct axios call
    expect(axios.post).toHaveBeenCalledWith(
      "http://localhost:5000/api/contact",
      {
        name: "Soham",
        email: "test@abcgmail.com",
        subject: "Subject of the Problem",
        message: "this is the description of problem",
      }
    );
  });
  test("matches snapshot", () => {
    const { asFragment } = render(<Contact />);
    expect(asFragment()).toMatchSnapshot();
  });
});
