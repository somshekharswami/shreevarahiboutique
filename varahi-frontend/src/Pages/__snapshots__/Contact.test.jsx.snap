// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Contact Page Form matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="min-h-screen bg-gradient-to-b from-pink-50 to-purple-50 py-12 px-6 md:px-20"
  >
    <div
      class="max-w-3xl mx-auto bg-white p-8 rounded-xl shadow-lg"
    >
      <form
        class="space-y-6"
      >
        <input
          aria-label="name"
          class="w-full border border-gray-300 px-4 py-2 rounded-md focus:ring-2 focus:ring-[#B19A99] outline-none font-mono"
          name="name"
          placeholder="Enter your full name"
          type="text"
        />
        <input
          aria-label="email"
          class="w-full border border-gray-300 px-4 py-2 rounded-md focus:ring-2 focus:ring-[#B19A99] outline-none font-mono"
          name="email"
          placeholder="abc@gmail.com"
          type="email"
        />
        <input
          aria-label="subject"
          class="w-full border border-gray-300 px-4 py-2 rounded-md focus:ring-2 focus:ring-[#B19A99] outline-none font-mono"
          name="subject"
          placeholder="Let us know why you're reaching out"
          type="text"
        />
        <textarea
          aria-label="message"
          class="w-full border border-gray-300 px-4 py-2 rounded-md resize-none focus:ring-2 focus:ring-[#B19A99] outline-none font-mono"
          name="message"
          placeholder="Briefly describe.."
          rows="5"
        />
        <div
          class="text-center"
        >
          <button
            aria-label="submit"
            class="bg-[#B19A99] text-white px-6 py-2 rounded-md hover:bg-[#a88a89] transition duration-300"
            type="submit"
          >
            Send Message
          </button>
        </div>
      </form>
    </div>
  </section>
</DocumentFragment>
`;
